src/app/loadout-builder/types.ts

```diff
@@ -91,11 +92,10 @@ export interface LockedArmor2Mod {
   /** Essentially an identifier for each mod, as a single mod definition can be selected multiple times.*/
   key?: number;
   modDef: PluggableInventoryItemDefinition;
-  category: ModPickerCategory;
 }
 
 export type LockedArmor2ModMap = {
-  [T in ModPickerCategory]: LockedArmor2Mod[];
+  [plugCategoryHash: number]: LockedArmor2Mod[] | undefined;
```

Comment:
This is the crux of the change, everything else is a flow on effect from this changing.